lock-api/src/main/java/com/palantir/lock/client/LockRefresher.java

```diff
@@ -70,8 +70,10 @@ private void refreshLocks() {
             if (!refreshFailures.isEmpty()) {
```

Comment:
On a related note, does this logging ever get racey? TimelockClient#unlock calls this first and then unlock on timelock, so if the refresher is currently refreshing, it won't see the fact that we just unlocked it.

Reply:
Basically this logline suggests either a leader election or something going weird with timelock, whereas it seems to me that it could fail to refresh if the lock was unlocked in between.

Reply:
I guess it might depend on how timelock unlocks this stuff

Reply:
Yep, it does. I had a brief look at the log spam last week, though didn't really get further than "async unlocker and refresher probably have a race condition"

Reply:
cool, so it does sound like this logline might be incorrect and spammy. I would vote to remove it, instead the more useful logline would probably be something along the lines of "Change of leader detected", because that's roughtly what you'd like to track.

Reply:
We will also hopefully now see if these are normal transaction locks that are being lost or immutable timestamp locks.